### PR TITLE
Fix panic in API pulls when headbranch does not exist

### LINKED ISSUE #10639: Bug: Api: ListPullRequests [500]
`curl -X GET "https://gitea.com/api/v1/repos/gitea/tea/pulls?state=closed&milestone=538&page=1" -H "accept: application/json"`

since no trace or debug info is shown and I cant reproduce on my test instance I cant veryfy or determine the reason :/

### PR DESCRIPTION
**Pull request by zeripath**
When a branch of a forked repository is merged as a PR and then deleted `/api/v1/repos/owner/name/pulls` and `/api/v1/repos/owner/name/pulls/index` will cause a 500 with a panic internally. 

This is because:

* modules/git/repo_commit.go:21:`GetRefCommitID(name string)` does not return a `git.ErrNotExist` error
* modules/convert/pull.go:110:`log.Error("GetBranch[%s]: %v", headBranch.Name, err)` causes a NPE because headBranch is nil as determined a few lines earlier.

These are now fixed.

Fix #10639 

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by zeripath** [comment]
I'm fairly certain that 1.11 will be affected too.

File: go.sum [hunk 1]
```diff
@@ -576,6 +576,7 @@ github.com/yohcop/openid-go v0.0.0-20160914080427-2c050d2dae53 h1:HsIQ6yAjfjQ3Ix
 github.com/yohcop/openid-go v0.0.0-20160914080427-2c050d2dae53/go.mod h1:f6elajwZV+xceiaqgRL090YzLEDGSbqr3poGL3ZgXYo=
 github.com/yuin/goldmark v1.1.24 h1:K4FemPDr4x/ZcqldoXWnexTLfdMIy2eEfXxsLnotTRI=
 github.com/yuin/goldmark v1.1.24/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9decYSb74=
+github.com/yuin/goldmark v1.1.25 h1:isv+Q6HQAmmL2Ofcmg8QauBmDPlUUnSoNhEcC940Rds=
```
Comment (lafriks) [inline go.sum]: Why this change?

Comment (zeripath) [reply go.sum]: because it keeps changing for me - I suspect if you check your go.mod this change will be happening for you too!

Comment (zeripath) [reply go.sum]: in fact if you run go mod tidy there's another change.

```diff
diff --git a/go.sum b/go.sum
index 1b75c6a71..505dd5571 100644
--- a/go.sum
+++ b/go.sum
@@ -574,8 +574,7 @@ github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2/go.mod h1:UETIi67q
 github.com/xordataexchange/crypt v0.0.3-0.20170626215501-b2862e3d0a77/go.mod h1:aYKd//L2LvnjZzWKhF00oedf
 github.com/yohcop/openid-go v0.0.0-20160914080427-2c050d2dae53 h1:HsIQ6yAjfjQ3IxPGrTusxp6Qxn92gNVq2x5Cbv
 github.com/yohcop/openid-go v0.0.0-20160914080427-2c050d2dae53/go.mod h1:f6elajwZV+xceiaqgRL090YzLEDGSbq
-github.com/yuin/goldmark v1.1.24 h1:K4FemPDr4x/ZcqldoXWnexTLfdMIy2eEfXxsLnotTRI=
-github.com/yuin/goldmark v1.1.24/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9decYSb74=
+github.com/yuin/goldmark v1.1.25 h1:isv+Q6HQAmmL2Ofcmg8QauBmDPlUUnSoNhEcC940Rds=
 github.com/yuin/goldmark v1.1.25/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9decYSb74=
 github.com/ziutek/mymysql v1.5.4 h1:GB0qdRGsTwQSBVYuVShFBKaXSnSnYYC2d9knnE1LHFs=
 github.com/ziutek/mymysql v1.5.4/go.mod h1:LMSpPZ6DbqWFxNCHW77HeMg9I646SAhApZ/wKdgO/C0=
```

Comment (zeripath) [reply go.sum]: I'll remove it and open a go mod tidy pr.

```suggestion
```

Comment (lafriks) [reply go.sum]: Yes it did for me also but this should be in other PR than

Comment (zeripath) [reply go.sum]: https://github.com/go-gitea/gitea/pull/10677

**Review by lafriks — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

File: modules/git/repo_commit.go [hunk 2]
```diff
@@ -21,6 +21,11 @@ import (
 func (repo *Repository) GetRefCommitID(name string) (string, error) {
 	ref, err := repo.gogitRepo.Reference(plumbing.ReferenceName(name), true)
 	if err != nil {
+		if err == plumbing.ErrReferenceNotFound {
```
Comment (lunny) [inline modules/git/repo_commit.go:24]: Are there any places to check if the error is `plumbing.ErrReferenceNotFound`?

Comment (zeripath) [reply modules/git/repo_commit.go:24]: Only one other place that I have changed to use the git error.
